File: internal/model/ai_session.go
package model

type AISessionIndex struct {
	UserID       string `gorm:"column:user_id;primaryKey"`
	ProviderID   string `gorm:"column:provider_id;primaryKey;index:idx_ai_session_provider_active"`
	SourcePath   string `gorm:"column:source_path;primaryKey;type:text"`
	SessionID    string `gorm:"column:session_id;index"`
	Title        string `gorm:"column:title"`
	Summary      string `gorm:"column:summary;type:text"`
	ProjectDir   string `gorm:"column:project_dir;type:text"`
	CreatedAt    int64  `gorm:"column:created_at"`
	LastActiveAt int64  `gorm:"column:last_active_at;index:idx_ai_session_provider_active"`
	MessageCount int    `gorm:"column:message_count"`
	ParseError   string `gorm:"column:parse_error;type:text"`
	FileSize     int64  `gorm:"column:file_size"`
	FileModTime  int64  `gorm:"column:file_mod_time"`
	ScannedAt    int64  `gorm:"column:scanned_at;index"`
}

func (AISessionIndex) TableName() string {
	return "ai_session_indices"
}

File: ui/src/types/ai-session.ts
export type AIProviderId = "claude" | "codex" | "gemini" | "opencode" | "openclaw";

export interface AIProviderConfig {
  enabled: boolean;
  paths: string[];
}

export interface AISessionConfig {
  providers: Record<string, AIProviderConfig>;
  autoRescanOnOpen: boolean;
  cacheEnabled: boolean;
  showParseErrors: boolean;
}

export interface AISessionMeta {
  providerId: string;
  sessionId: string;
  title?: string;
  summary?: string;
  projectDir?: string;
  createdAt?: number;
  lastActiveAt?: number;
  sourcePath: string;
  messageCount?: number;
  parseError?: string;
  fileSize?: number;
  fileModTime?: number;
  scannedAt?: number;
}

export interface AISessionMessage {
  role: string;
  content: string;
  ts?: number;
}

export interface AIProviderStatus {
  providerId: string;
  enabled: boolean;
  paths: string[];
  available: boolean;
  sessionCount: number;
  errorCount: number;
  lastScanAt?: number;
}

export interface AIOverviewResponse {
  totalSessions: number;
  enabledProviders: number;
  scannedAt?: number;
  fromCache: boolean;
  providerStatus: AIProviderStatus[];
}

export interface AIListResponse {
  sessions: AISessionMeta[];
  providerStatus: AIProviderStatus[];
  fromCache: boolean;
  scannedAt?: number;
  config: AISessionConfig;
}

export interface AIMessagesResponse {
  session: AISessionMeta;
  messages: AISessionMessage[];
  parseWarnings: string[];
}

export const AI_PROVIDER_ORDER: AIProviderId[] = ["claude", "codex", "gemini", "opencode", "openclaw"];

export const AI_PROVIDER_TONE: Record<AIProviderId, string> = {
  claude: "bg-orange-500/12 text-orange-500 border-orange-500/20",
  codex: "bg-emerald-500/12 text-emerald-500 border-emerald-500/20",
  gemini: "bg-sky-500/12 text-sky-500 border-sky-500/20",
  opencode: "bg-amber-500/12 text-amber-600 border-amber-500/20",
  openclaw: "bg-rose-500/12 text-rose-500 border-rose-500/20",
};
